misk-zookeeper/src/main/kotlin/misk/clustering/zookeeper/ZkLeaseManager.kt

```diff
@@ -65,19 +66,17 @@ internal class ZkLeaseManager @Inject internal constructor(
     if (!state.compareAndSet(State.NOT_STARTED, State.RUNNING)) return
     connected.set(false)
 
-    // When a cluster change occurs, recheck each lease to see if it should still be owned
-    cluster.watch {
-      actionQueue.put {
-        handleClusterChange()
-      }
-    }
-
     // When we are disconnected from zk, inform all of the leases that their state is now unknown
     client.value.connectionStateListenable.addListener(ConnectionStateListener { _, state ->
       actionQueue.put {
         handleConnectionStateChanged(state.isConnected)
       }
     })
+
+    taskQueue.schedule(leasePollInterval) {
```

Comment:
Downside here is that we've made lease checking sequential whereas before on cluster change we would check for changes in ownership for each lease in parallel. Probably doesn't matter much given that only a small number of leases are likely to change ownership state (thus necessitating calls to zk) on each pass.

Reply:
That's what I was thinking too. I could have kept the cached thread executor, but since it's polling every second and there's unlikely to be much i/o, I figured sequential is better.

Reply:
my first thought is we just schedule a task for every lease. however, the set of leases is not static, so that's kind of tricky.